### PR TITLE
fix(core): normalizeRoutePath remove only known extensions

### LINKED ISSUE #954: [Bug]: links with `.` (periods) are reported as broken
### Version

```sh
System:
    OS: macOS 14.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 39.19 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 123.0.6312.123
    Safari: 17.1
  npmPackages:
    rspress: ^1.18.1 => 1.18.1
```

### Details

When using `typedoc`, it can generate names of files with periods in them like this:
- `classes/plugins.OutputPlugin`
- `classes/plugins.LoggerPlugin`
then the filename gets put into the route path and despite the links working, they will get reported as being broken when using `markdown.checkDeadLinks` option in the config

### Reproduce link

https://github.com/callstack/repack/tree/docs/migration

### Reproduce Steps

1. `pnpm install`
2. `cd new-website`
3. `pnpm start`

### PR DESCRIPTION
**Pull request by jbroma**
## Summary

- remove only known extensions from route paths
- removed redundant `normalizeRoutePath` from `routeData`
- expose `normalizeRoutePath` from `RouteService`

## Related Issue

fix #954 

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Comment by Timeless0911** [comment]
LGTM

**Review by Timeless0911 — DISMISSED** [review]

File: packages/core/src/node/mdx/remarkPlugins/checkDeadLink.ts [hunk 1]
```diff
@@ -66,9 +65,9 @@ export const remarkCheckDeadLinks: Plugin<
       }
 
       if (!url.startsWith('http') && !url.startsWith('https')) {
-        const normalizeUrl = normalizeRoutePath(
+        const { routePath: normalizeUrl } = routeService.normalizeRoutePath(
           // fix: windows path
-          url?.split(path.sep).join('/')?.split('#')[0],
+          url.split(path.sep).join('/')?.split('#')[0],
```
Comment (jbroma) [inline packages/core/src/node/mdx/remarkPlugins/checkDeadLink.ts:70]: FYI, I've removed optional operator here as from TS definitions it didn't seem to be needed

**Comment by sanyuan0704** [comment]
Why do you want to put the implement of normalizeRoutePath into `RouteService` instead of acting as a universal util function? @jbroma

**Comment by jbroma** [comment]
> Why do you want to put the implement of normalizeRoutePath into `RouteService` instead of acting as a universal util function? @jbroma

@sanyuan0704 The reason `normalizeRoutePath` was put inside of `RouteService` was to access the list of extensions from the user supplied settings. We could keep it outside but then we would need to rely on default extensions only.

**Comment by sanyuan0704** [comment]
> > Why do you want to put the implement of normalizeRoutePath into `RouteService` instead of acting as a universal util function? @jbroma
> 
> @sanyuan0704 The reason `normalizeRoutePath` was put inside of `RouteService` was to access the list of extensions from the user supplied settings. We could keep it outside but then we would need to rely on default extensions only.

@jbroma You can add extensions param in `normalizeRoutePath` and call it in the private method of RouteService.

**Comment by jbroma** [comment]
@sanyuan0704 oh right, that makes sense, I'll rework this PR so it's not a breaking change like this, forgot this are not internal APIs, thanks for spotting that!

**Comment by jbroma** [comment]
@sanyuan0704 upon closer inspection these functions were not exposed as public API outside of core package so they wouldn't be breaking (only `normalizeRoutePath` from `rspress/runtime` is public)

I've brough back `normalizeRoutePath` outside of the `RouteService` but also made the previously private `#normalizeRoutePath` public so that we can use the extensions from `RouteService` inside of `checkDeadLinks`.

Let me know whether this is ok now

**Review by sanyuan0704 — APPROVED** [review]

**Pull request merged**